packages/tree/src/lib/TreeItem.tsx

```diff
@@ -5,18 +5,29 @@ export interface TreeItemProps extends MuiTreeItem2Props {
   truncate?: boolean;
 }
 
-export const TreeItem = ({ children, slots, slotProps, truncate, ...rest }: TreeItemProps): JSX.Element => {
+export const TreeItem = ({ children, label, slots, slotProps, truncate, ...rest }: TreeItemProps): JSX.Element => {
   return (
     <TreeItem2
       {...rest}
-      slots={{ ...slots, expandIcon: TriangleRightIcon, collapseIcon: TriangleExpandIcon }}
+      label={label}
+      slots={{
+        ...slots,
+        expandIcon: TriangleRightIcon,
+        collapseIcon: TriangleExpandIcon,
+      }}
       slotProps={{
         ...slotProps,
         expandIcon: { fontSize: 'Xsmall' },
         collapseIcon: { fontSize: 'Xsmall' },
         label: {
           style: truncate ? { overflow: 'hidden', textOverflow: 'ellipsis', whiteSpace: 'nowrap' } : {},
         },
+        checkbox: {
+          // @ts-expect-error TreeItem2SlotProps.checkbox types are incorrect
+          inputProps: {
+            'aria-label': `${label} Checkbox`,
```

Comment:
this might require some more research, but I'm not sure the checkbox should be visible to assistive technologies at all which negates the need for an aria-label. 
The parent treeitem has the aria-selected property, and for keyboard users spacebar on the treeitem selects it vs having to focus the checkbox itself.

Reply:
hmmm... that's a good point. If use the user, you have to click on the checkbox. Clicking the list item does not select it. This might be a weird situation.

Reply:
the more i look/use this the more I believe the checkbox input should be aria-hidden. If it's not hidden then the label for the treeitem ends up being {checkbox label}{label} so it's repetitive. moreover, if you use AT to force focus on the checkbox and then select it with the keyboard it breaks a lot of the functionality when moving back to the treeitem. We could go through the trouble of manually assigning an aria-label to the treeitem so it doesn't have the repetitive children name, but at that point what else will we run into?

Looking at the keyboard interactions for treenode (https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/treeitem_role#keyboard_interactions) it also makes sense to not have an additional checkbox for keyboard users to interact with.

Reply:
That makes sense. Thank you for looking into it!